### PR TITLE
feat: Add CasbinFairing

### PR DESCRIPTION
**Pull request by uran0sH**
I have completed a basic CasbinFairing.

### CODE REVIEW

**Comment by hackerchai** [comment]
You may add the test coverage to ensure the availablity

**Comment by uran0sH** [comment]
@hackerchai I have added test for fairing. plz review.

**Comment by hackerchai** [comment]
> @hackerchai I have added test for fairing. plz review.

also with github ci

File: src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        }
+    }
+
+    fn on_request(&self, request: &mut Request, _data: &Data) {
+        let cloned_enforce = self.enforcer.clone();
+        let path = request.uri().path().to_owned();
+        let action = request.method().as_str().to_owned();
+        // Get subject and domain from cookie.
+        let subject = request.cookies().get("subject").map(|x| x.to_owned());
+        let domain = request.cookies().get("domain").map(|x| x.to_owned());
```
Comment (hackerchai) [inline src/lib.rs]: we shoud not encode the `subject` and `domain` into cookie

File: tests/test_fairing.rs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    routes,
+};
+use rocket_authz;
+
+#[get("/login")]
+fn login(mut cookies: Cookies) -> &'static str {
+    cookies.add(Cookie::new("subject", "alice"));
+    "success"
+}
+
```
Comment (hackerchai) [inline tests/test_fairing.rs]: we should also cover `domain` test

**Review by hackerchai — COMMENTED** [review]
please fix these

File: src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,128 @@
+use rocket::{
+    fairing::{Fairing, Info, Kind},
+    http::Status,
+    request::{self, FromRequest, Request},
+    Data,
+};
+
+use casbin::prelude::*;
+use casbin::{CachedEnforcer, CoreApi, Result as CasbinResult};
+use std::sync::{Arc, RwLock};
```
Comment (hackerchai) [inline src/lib.rs]: please use the async cversion

Comment (PsiACE) [inline src/lib.rs]: Compared to `std::sync::RwLock` , `parking_lot::RwLock` is more recommended.

**Pull request merged**